lib/Service/RoomService.php

```diff
@@ -416,6 +427,83 @@ public function setCallRecording(Room $room, int $status = Room::RECORDING_NONE)
 		$this->dispatcher->dispatch(Room::EVENT_AFTER_SET_CALL_RECORDING, $event);
 	}
 
+	public function storeRecording(Room $room, string $owner, array $file, IRootFolder $rootFolder): void {
+		if (is_null($file) || !is_array($file)) {
+			throw new InvalidArgumentException('no_file');
+		}
+
+		if (
+			$file['error'] !== 0 ||
+			!is_uploaded_file($file['tmp_name']) ||
+			Filesystem::isFileBlacklisted($file['tmp_name'])
+		) {
+			throw new InvalidArgumentException('invalid_file');
+		}
+
+		$content = file_get_contents($file['tmp_name']);
+		unlink($file['tmp_name']);
+
+		$mimeType = $this->mimeTypeDetector->detectString($content);
+		$allowedMimeTypes = [
+			'video/mp4',
+			'video/mpeg',
+			'video/ogg',
+			'audio/mp3',
+			'audio/ogg',
```

Comment:
@nickvergessen  is this format list ok? Do you think that have a good place to document this list?

Reply:
Solved making the list configurable.